versioned/persist/rocks/src/main/java/org/projectnessie/versioned/persist/rocks/RocksDatabaseAdapter.java

```diff
@@ -322,4 +327,45 @@ protected int entitySize(CommitLogEntry entry) {
   protected int entitySize(KeyWithType entry) {
     return toProto(entry).getSerializedSize();
   }
+
+  @Override
+  protected void writeRefLog(NonTransactionalOperationContext ctx, AdapterTypes.RefLogEntry entry)
+      throws ReferenceConflictException {
+    Lock lock = dbInstance.getLock().writeLock();
```

Comment:
I'm a little worried that we're acquiring the lock separately for the ref log entry. If there's a commit done, the commit and reflog operations will take locks independent of each other, if commit operation passes, whereas reflog insert fails, it's gonna leave an inconsistent state.

Reply:
multi table transaction is handled by CAS operation loop present in the caller NonTransactionalDatabaseAdapter#casOpLoop.  

which will takes care of reverting all the entries when any one of the commit failed.